fpdf/fpdf.py

```diff
@@ -121,6 +121,43 @@ class ToCPlaceholder(NamedTuple):
     pages: int = 1
 
 
+class SubsetMap:
+    """Holds a mapping of used characters and their position in the font's subset
+
+    Characters that must be mapped on their actual unicode must be part of the
+    `identities` list during object instanciation. `pick()` can be used to get the
+    characters corresponding position in the subset. If it's not yet part of the
+    object, a new position is acquired automatically. This implementation always
+    tries to return the lowest possible representation.
+    """
+
+    def __init__(self, identities: List[int]):
+        super().__init__()
+        self._next = 0
+
+        # sort list to ease deletion once _next
+        # becomes higher than first reservation
+        self._reserved = sorted(identities)
```

Comment:
This really is an edge case, but `SubsetMap.pick` will no behave as expected if `identities` contain duplicates here, _e.g._ if `identities == (0, 0, 0, 2)`

Reply:
Indeed, I have not considered this case. The same applies to negative values. But apart from the fact that `_reserved` would not be cleared properly, it should actually have no noticeable impact. I will now point this out in the docstring and weakly type `identities` as `Set[int]`. `sbarr` will only hold acceptable values in practice anyway. Do you think this is enough or should it still be checked more aggressively?

Reply:
That's perfect for me, no need for more defensive programming here I think 😉